Model/RemotePageRepository.php

```diff
@@ -128,6 +128,25 @@ public function get(int $siteId, int $pageId): ?array
         return $page;
     }
 
+    public function postMetaData(int $siteId, int $pageId, string $key, string $value)
+    {
+        $site = $this->siteRepository->get($siteId);
+        $this->pageClient->setBaseUrl($site->getBaseurl());
+        $authentication = $site->getApiUsername().':'.$site->getApiPassword();
```

Comment:
What happens if API credentials are empty?

Reply:
Then there will be an exception whose error shall be logged. A notice about this will be shown to the user.

Reply:
You're better off checking it beforehand and informing earlier than letting the request fail.

Reply:
Check was added